### DISCUSSION TITLE
Relation 1-N control-measurements

### DISCUSSION THREAD

**Discussion by arnaud-lxbg** [dbarzin/deming#78] [Ideas]
Salut Didier,

Je viens de me lancer dans Deming, je découvre l'outil au fur et à mesure, le dernier podcast en ligne est très utile pour la prise en main de l'outil, peut-être devrais-tu faire plus de tutoriels en séquences vidéo.

J'ai une question concernant les choix d'implémentation, pourquoi avoir choisi une relation 1 à 1 entre un control et un measurement?
L'opportunité de définir plusieurs mesurements qui satisfassent un control me semblait pertinente, or je n'arrive pas à la mettre en place. J'imagine donc qu'il y a une relation 1 à 1 entre les deux.

De même, mais c'est certainement plus complexe à implémenter, un measurement devrait pouvoir contribuer à s'assurer de l'efficacité de plusieurs controls.

Il y a peut être quelque chose que je n'ai pas compris dans la mise en oeuvre, ou dans la philosophie de l'outil, mais le fait de devoir décliner les controls sur un même plan, sans qu'il existe de sous-controls me semble extrêmement complexe à gérer (besoin de trouver une numérotation spécifique...) et peu efficace pour le dashboarding.

Autre question liée, si la philosophie de l'outil est de multiplier les controls, pourquoi ne pas implémenter de fonction de copie qui permettrait de gagner un temps fou dans l'application? aujourd'hui cela est fait de manière très efficace par l'import excel, donc j'imagine que c'est uen question de priorité

En tout cas, bravo pour l'outil, il me semble très efficace même si je démarre dans son utilisation.

Arnaud S.

**Comment by charlesgoyard** [comment]
Je suis d'accord que ce serait pratique de pouvoir avoir un 1-N, voire même un N-N. Ceci dit une mesure de contournement est  discutée dans #34.

**Reply by arnaud-lxbg** [reply]
Merci, je ne l'avais pas vu, je viens de poster la suite dans la discussion 34.

A mon avis, la mesure de contournement peut être utile, mais ne répond pas totalement à la problèmatique.

**Reply by dbarzin** [reply]
L'idée serait donc d'ajouter un bouton "copier" sur l'écran d'affichage d'une mesure de sécurité qui renverrait vers l'écran de création d'une mesure de sécurité avec tous les champs du formulaire déjà remplis avec les valeurs de la mesure à copier.

**Reply by arnaud-lxbg** [reply]
Tant qu'on doit lier une mesure à un contrôle, oui ça serait l'idée.
A voir si ça ne serait pas utile aussi au niveau contrôle si on change le scope.

**Reply by dbarzin** [reply]
On peut déjà changer le périmètre d'un contrôle, je ne comprends pas la remarque.

**Reply by arnaud-lxbg** [reply]
Je m'explique:
pouvoir dupliquer un contrôle serait utile aussi, même si, pour pouvoir le valider, il faut, à minima, changer le scope, puisqu'on ne peut pas avoir deux contrôles sur le même scope.

**Reply by dbarzin** [reply]
:tada: C'est en place ! Il est maintenant possible de copier une mesure de sécurité.
